public/app/plugins/datasource/prometheus/querybuilder/shared/parsingUtils.ts

```diff
@@ -65,7 +65,7 @@ const varTypeFunc = [
  * Get back the text with variables in their original format.
  * @param expr
  */
-function returnVariables(expr: string) {
+export function returnVariables(expr: string) {
```

Comment:
Out of curiosity, does it makes sense to stay with this utility function here? We can have a top-level shared folder with such things and reuse them as needed in low-level system pieces. Or even ignore DRY, replicate the code on each Query Builder, and let it grow independently of the query builder. Why do we share parts of these Loki/Prom query builders, like in this case? 

Sorry for the loads of questions 😄 Just looking to understand this solution design decision between Loki/Prom Query Builders :)

Reply:
I don't have deep insights in the architectural decisions, but I would assume it's hard to pick a spot outside of both `datasources` directories where such code would live. I see your point, that Loki and Prometheus datasources are now coupled quite strongly together, but on the other hand they both are core datasources and deselecting some of those when building Grafana is also not possible.

I don't know if @ivanahuckova and @aocenas might have better answers to those questions ;)

Reply:
@obetomuniz great question! For now, this is actually a place for shared code between `loki` and `prometheus` query builder. Everything in `shared` could be in the future moved somewhere else. The idea is, that in the future, when we add new query builders (e.g. for `flux`) we would figure out where to move all shared logic (either `grafana/ui`, `grafana/data`, new package - `grafana/builder`). But for now, we use prometheus `shared` sub-folder for this.